File: src/ros2_bridge/mappers/std_msgs/int64.rs
//! Typed mapper for `std_msgs/msg/Int64`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn int64_to_bus(msg: ros_env::std_msgs::msg::Int64) -> crate::std_msgs::msg::v1::Int64 {
    crate::std_msgs::msg::v1::Int64 {
        data: msg.data,
    }
}

pub(crate) fn int64_to_ros(bus: crate::std_msgs::msg::v1::Int64) -> ros_env::std_msgs::msg::Int64 {
    ros_env::std_msgs::msg::Int64 {
        data: bus.data,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct StdMsgsInt64Mapper;

impl TypedTopicMapper for StdMsgsInt64Mapper {
    type Ros = ros_env::std_msgs::msg::Int64;
    type Bus = crate::std_msgs::msg::v1::Int64;

    fn type_name(&self) -> &'static str {
        "std_msgs/msg/Int64"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(int64_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(int64_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/std_msgs/int64.py
"""Generated mapper for `std_msgs/msg/Int64`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert


def int64_to_bus(msg):
    from robot_bus.std_msgs.msg.v1 import Int64 as BusMsg

    bus = BusMsg()
    bus.data = msg.data
    return bus


def int64_to_ros(bus):
    from std_msgs.msg import Int64 as RosMsg

    out = RosMsg()
    out.data = bus.data
    return out


class StdMsgsInt64Mapper:
    def type_name(self) -> str:
        return "std_msgs/msg/Int64"

    def ros_msg_type(self):
        from std_msgs.msg import Int64 as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return int64_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.std_msgs.msg.v1 import Int64 as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return int64_to_ros(bus)
